// klaviyo-api/dist/model/clientBISSubscriptionCreateQueryResourceObjectAttributes.js
"use strict";
/**
 * Klaviyo API
 * The Klaviyo REST API. Please visit https://developers.klaviyo.com for more details.
 *
 * Contact: developers@klaviyo.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ClientBISSubscriptionCreateQueryResourceObjectAttributes = void 0;
class ClientBISSubscriptionCreateQueryResourceObjectAttributes {
    static getAttributeTypeMap() {
        return ClientBISSubscriptionCreateQueryResourceObjectAttributes.attributeTypeMap;
    }
}
exports.ClientBISSubscriptionCreateQueryResourceObjectAttributes = ClientBISSubscriptionCreateQueryResourceObjectAttributes;
ClientBISSubscriptionCreateQueryResourceObjectAttributes.attributeTypeMap = [
    {
        "name": "channels",
        "baseName": "channels",
        "type": "Array<ClientBISSubscriptionCreateQueryResourceObjectAttributes.ChannelsEnum>"
    },
    {
        "name": "profile",
        "baseName": "profile",
        "type": "ClientBISSubscriptionCreateQueryResourceObjectAttributesProfile"
    }
];
(function (ClientBISSubscriptionCreateQueryResourceObjectAttributes) {
    let ChannelsEnum;
    (function (ChannelsEnum) {
        ChannelsEnum[ChannelsEnum["Email"] = 'EMAIL'] = "Email";
        ChannelsEnum[ChannelsEnum["Push"] = 'PUSH'] = "Push";
        ChannelsEnum[ChannelsEnum["Sms"] = 'SMS'] = "Sms";
    })(ChannelsEnum = ClientBISSubscriptionCreateQueryResourceObjectAttributes.ChannelsEnum || (ClientBISSubscriptionCreateQueryResourceObjectAttributes.ChannelsEnum = {}));
})(ClientBISSubscriptionCreateQueryResourceObjectAttributes = exports.ClientBISSubscriptionCreateQueryResourceObjectAttributes || (exports.ClientBISSubscriptionCreateQueryResourceObjectAttributes = {}));
