// klaviyo-api/dist/model/dataSourceCreateQueryResourceObjectAttributes.js
"use strict";
/**
 * Klaviyo API
 * The Klaviyo REST API. Please visit https://developers.klaviyo.com for more details.
 *
 * Contact: developers@klaviyo.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.DataSourceCreateQueryResourceObjectAttributes = void 0;
class DataSourceCreateQueryResourceObjectAttributes {
    constructor() {
        this['description'] = '';
    }
    static getAttributeTypeMap() {
        return DataSourceCreateQueryResourceObjectAttributes.attributeTypeMap;
    }
}
exports.DataSourceCreateQueryResourceObjectAttributes = DataSourceCreateQueryResourceObjectAttributes;
DataSourceCreateQueryResourceObjectAttributes.attributeTypeMap = [
    {
        "name": "title",
        "baseName": "title",
        "type": "string"
    },
    {
        "name": "visibility",
        "baseName": "visibility",
        "type": "DataSourceCreateQueryResourceObjectAttributes.VisibilityEnum"
    },
    {
        "name": "description",
        "baseName": "description",
        "type": "string"
    }
];
(function (DataSourceCreateQueryResourceObjectAttributes) {
    let VisibilityEnum;
    (function (VisibilityEnum) {
        VisibilityEnum[VisibilityEnum["Private"] = 'private'] = "Private";
        VisibilityEnum[VisibilityEnum["Shared"] = 'shared'] = "Shared";
    })(VisibilityEnum = DataSourceCreateQueryResourceObjectAttributes.VisibilityEnum || (DataSourceCreateQueryResourceObjectAttributes.VisibilityEnum = {}));
})(DataSourceCreateQueryResourceObjectAttributes = exports.DataSourceCreateQueryResourceObjectAttributes || (exports.DataSourceCreateQueryResourceObjectAttributes = {}));
