// klaviyo-api/dist/model/dataSourceResponseObjectResourceAttributes.js
"use strict";
/**
 * Klaviyo API
 * The Klaviyo REST API. Please visit https://developers.klaviyo.com for more details.
 *
 * Contact: developers@klaviyo.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.DataSourceResponseObjectResourceAttributes = void 0;
class DataSourceResponseObjectResourceAttributes {
    static getAttributeTypeMap() {
        return DataSourceResponseObjectResourceAttributes.attributeTypeMap;
    }
}
exports.DataSourceResponseObjectResourceAttributes = DataSourceResponseObjectResourceAttributes;
DataSourceResponseObjectResourceAttributes.attributeTypeMap = [
    {
        "name": "title",
        "baseName": "title",
        "type": "string"
    },
    {
        "name": "visibility",
        "baseName": "visibility",
        "type": "DataSourceResponseObjectResourceAttributes.VisibilityEnum"
    },
    {
        "name": "description",
        "baseName": "description",
        "type": "string"
    }
];
(function (DataSourceResponseObjectResourceAttributes) {
    let VisibilityEnum;
    (function (VisibilityEnum) {
        VisibilityEnum[VisibilityEnum["Private"] = 'private'] = "Private";
        VisibilityEnum[VisibilityEnum["Shared"] = 'shared'] = "Shared";
    })(VisibilityEnum = DataSourceResponseObjectResourceAttributes.VisibilityEnum || (DataSourceResponseObjectResourceAttributes.VisibilityEnum = {}));
})(DataSourceResponseObjectResourceAttributes = exports.DataSourceResponseObjectResourceAttributes || (exports.DataSourceResponseObjectResourceAttributes = {}));
