// klaviyo-api/dist/model/flowWhatsApp.js
"use strict";
/**
 * Klaviyo API
 * The Klaviyo REST API. Please visit https://developers.klaviyo.com for more details.
 *
 * Contact: developers@klaviyo.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.FlowWhatsApp = void 0;
class FlowWhatsApp {
    constructor() {
        this['smartSendingEnabled'] = true;
        this['transactional'] = false;
        this['addTrackingParams'] = false;
    }
    static getAttributeTypeMap() {
        return FlowWhatsApp.attributeTypeMap;
    }
}
exports.FlowWhatsApp = FlowWhatsApp;
FlowWhatsApp.attributeTypeMap = [
    {
        "name": "id",
        "baseName": "id",
        "type": "string"
    },
    {
        "name": "name",
        "baseName": "name",
        "type": "string"
    },
    {
        "name": "vendorId",
        "baseName": "vendor_id",
        "type": "string"
    },
    {
        "name": "smartSendingEnabled",
        "baseName": "smart_sending_enabled",
        "type": "boolean"
    },
    {
        "name": "transactional",
        "baseName": "transactional",
        "type": "boolean"
    },
    {
        "name": "addTrackingParams",
        "baseName": "add_tracking_params",
        "type": "boolean"
    },
    {
        "name": "customTrackingParams",
        "baseName": "custom_tracking_params",
        "type": "Array<UtmParam>"
    },
    {
        "name": "additionalFilters",
        "baseName": "additional_filters",
        "type": "ConditionalBranchActionDataProfileFilter"
    }
];
