// klaviyo-api/dist/model/mappedMetricResponseObjectResource.js
"use strict";
/**
 * Klaviyo API
 * The Klaviyo REST API. Please visit https://developers.klaviyo.com for more details.
 *
 * Contact: developers@klaviyo.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.MappedMetricResponseObjectResource = void 0;
class MappedMetricResponseObjectResource {
    static getAttributeTypeMap() {
        return MappedMetricResponseObjectResource.attributeTypeMap;
    }
}
exports.MappedMetricResponseObjectResource = MappedMetricResponseObjectResource;
MappedMetricResponseObjectResource.attributeTypeMap = [
    {
        "name": "type",
        "baseName": "type",
        "type": "MappedMetricEnum"
    },
    {
        "name": "id",
        "baseName": "id",
        "type": "MappedMetricResponseObjectResource.IdEnum"
    },
    {
        "name": "attributes",
        "baseName": "attributes",
        "type": "MappedMetricResponseObjectResourceAttributes"
    },
    {
        "name": "links",
        "baseName": "links",
        "type": "ObjectLinks"
    }
];
(function (MappedMetricResponseObjectResource) {
    let IdEnum;
    (function (IdEnum) {
        IdEnum[IdEnum["AddedToCart"] = 'added_to_cart'] = "AddedToCart";
        IdEnum[IdEnum["CancelledSales"] = 'cancelled_sales'] = "CancelledSales";
        IdEnum[IdEnum["OrderedProduct"] = 'ordered_product'] = "OrderedProduct";
        IdEnum[IdEnum["RefundedSales"] = 'refunded_sales'] = "RefundedSales";
        IdEnum[IdEnum["Revenue"] = 'revenue'] = "Revenue";
        IdEnum[IdEnum["StartedCheckout"] = 'started_checkout'] = "StartedCheckout";
        IdEnum[IdEnum["ViewedProduct"] = 'viewed_product'] = "ViewedProduct";
    })(IdEnum = MappedMetricResponseObjectResource.IdEnum || (MappedMetricResponseObjectResource.IdEnum = {}));
})(MappedMetricResponseObjectResource = exports.MappedMetricResponseObjectResource || (exports.MappedMetricResponseObjectResource = {}));
