// klaviyo-api/dist/model/predictiveAnalytics.js
"use strict";
/**
 * Klaviyo API
 * The Klaviyo REST API. Please visit https://developers.klaviyo.com for more details.
 *
 * Contact: developers@klaviyo.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PredictiveAnalytics = void 0;
class PredictiveAnalytics {
    static getAttributeTypeMap() {
        return PredictiveAnalytics.attributeTypeMap;
    }
}
exports.PredictiveAnalytics = PredictiveAnalytics;
PredictiveAnalytics.attributeTypeMap = [
    {
        "name": "historicClv",
        "baseName": "historic_clv",
        "type": "number"
    },
    {
        "name": "predictedClv",
        "baseName": "predicted_clv",
        "type": "number"
    },
    {
        "name": "totalClv",
        "baseName": "total_clv",
        "type": "number"
    },
    {
        "name": "historicNumberOfOrders",
        "baseName": "historic_number_of_orders",
        "type": "number"
    },
    {
        "name": "predictedNumberOfOrders",
        "baseName": "predicted_number_of_orders",
        "type": "number"
    },
    {
        "name": "averageDaysBetweenOrders",
        "baseName": "average_days_between_orders",
        "type": "number"
    },
    {
        "name": "averageOrderValue",
        "baseName": "average_order_value",
        "type": "number"
    },
    {
        "name": "churnProbability",
        "baseName": "churn_probability",
        "type": "number"
    },
    {
        "name": "expectedDateOfNextOrder",
        "baseName": "expected_date_of_next_order",
        "type": "Date"
    },
    {
        "name": "rankedChannelAffinity",
        "baseName": "ranked_channel_affinity",
        "type": "Array<PredictiveAnalytics.RankedChannelAffinityEnum>"
    }
];
(function (PredictiveAnalytics) {
    let RankedChannelAffinityEnum;
    (function (RankedChannelAffinityEnum) {
        RankedChannelAffinityEnum[RankedChannelAffinityEnum["Email"] = 'email'] = "Email";
        RankedChannelAffinityEnum[RankedChannelAffinityEnum["Push"] = 'push'] = "Push";
        RankedChannelAffinityEnum[RankedChannelAffinityEnum["Sms"] = 'sms'] = "Sms";
    })(RankedChannelAffinityEnum = PredictiveAnalytics.RankedChannelAffinityEnum || (PredictiveAnalytics.RankedChannelAffinityEnum = {}));
})(PredictiveAnalytics = exports.PredictiveAnalytics || (exports.PredictiveAnalytics = {}));
