// klaviyo-api/dist/model/profileMetricCondition.js
"use strict";
/**
 * Klaviyo API
 * The Klaviyo REST API. Please visit https://developers.klaviyo.com for more details.
 *
 * Contact: developers@klaviyo.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ProfileMetricCondition = void 0;
class ProfileMetricCondition {
    static getAttributeTypeMap() {
        return ProfileMetricCondition.attributeTypeMap;
    }
}
exports.ProfileMetricCondition = ProfileMetricCondition;
ProfileMetricCondition.attributeTypeMap = [
    {
        "name": "type",
        "baseName": "type",
        "type": "ProfileMetricEnum"
    },
    {
        "name": "metricId",
        "baseName": "metric_id",
        "type": "string"
    },
    {
        "name": "measurement",
        "baseName": "measurement",
        "type": "ProfileMetricCondition.MeasurementEnum"
    },
    {
        "name": "measurementFilter",
        "baseName": "measurement_filter",
        "type": "NumericOperatorFilter"
    },
    {
        "name": "timeframeFilter",
        "baseName": "timeframe_filter",
        "type": "ProfileMetricConditionTimeframeFilter"
    },
    {
        "name": "metricFilters",
        "baseName": "metric_filters",
        "type": "Array<ProfileMetricPropertyFilter>"
    }
];
(function (ProfileMetricCondition) {
    let MeasurementEnum;
    (function (MeasurementEnum) {
        MeasurementEnum[MeasurementEnum["Count"] = 'count'] = "Count";
        MeasurementEnum[MeasurementEnum["Sum"] = 'sum'] = "Sum";
    })(MeasurementEnum = ProfileMetricCondition.MeasurementEnum || (ProfileMetricCondition.MeasurementEnum = {}));
})(ProfileMetricCondition = exports.ProfileMetricCondition || (exports.ProfileMetricCondition = {}));
