// klaviyo-api/dist/model/profilePredictiveAnalyticsChannelAffinityRankCondition.js
"use strict";
/**
 * Klaviyo API
 * The Klaviyo REST API. Please visit https://developers.klaviyo.com for more details.
 *
 * Contact: developers@klaviyo.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ProfilePredictiveAnalyticsChannelAffinityRankCondition = void 0;
class ProfilePredictiveAnalyticsChannelAffinityRankCondition {
    static getAttributeTypeMap() {
        return ProfilePredictiveAnalyticsChannelAffinityRankCondition.attributeTypeMap;
    }
}
exports.ProfilePredictiveAnalyticsChannelAffinityRankCondition = ProfilePredictiveAnalyticsChannelAffinityRankCondition;
ProfilePredictiveAnalyticsChannelAffinityRankCondition.attributeTypeMap = [
    {
        "name": "type",
        "baseName": "type",
        "type": "ProfilePredictiveAnalyticsEnum"
    },
    {
        "name": "dimension",
        "baseName": "dimension",
        "type": "ProfilePredictiveAnalyticsChannelAffinityRankCondition.DimensionEnum"
    },
    {
        "name": "measurement",
        "baseName": "measurement",
        "type": "RankEnum"
    },
    {
        "name": "predictedChannel",
        "baseName": "predicted_channel",
        "type": "ProfilePredictiveAnalyticsChannelAffinityRankCondition.PredictedChannelEnum"
    },
    {
        "name": "filter",
        "baseName": "filter",
        "type": "ProfilePredictiveAnalyticsChannelAffinityRankFilter"
    }
];
(function (ProfilePredictiveAnalyticsChannelAffinityRankCondition) {
    let DimensionEnum;
    (function (DimensionEnum) {
        DimensionEnum[DimensionEnum["ChannelAffinity"] = 'channel_affinity'] = "ChannelAffinity";
    })(DimensionEnum = ProfilePredictiveAnalyticsChannelAffinityRankCondition.DimensionEnum || (ProfilePredictiveAnalyticsChannelAffinityRankCondition.DimensionEnum = {}));
    let PredictedChannelEnum;
    (function (PredictedChannelEnum) {
        PredictedChannelEnum[PredictedChannelEnum["Email"] = 'email'] = "Email";
        PredictedChannelEnum[PredictedChannelEnum["Push"] = 'push'] = "Push";
        PredictedChannelEnum[PredictedChannelEnum["Sms"] = 'sms'] = "Sms";
    })(PredictedChannelEnum = ProfilePredictiveAnalyticsChannelAffinityRankCondition.PredictedChannelEnum || (ProfilePredictiveAnalyticsChannelAffinityRankCondition.PredictedChannelEnum = {}));
})(ProfilePredictiveAnalyticsChannelAffinityRankCondition = exports.ProfilePredictiveAnalyticsChannelAffinityRankCondition || (exports.ProfilePredictiveAnalyticsChannelAffinityRankCondition = {}));
