// klaviyo-api/dist/model/profilePredictiveAnalyticsChannelAffinityPriorityCondition.js
"use strict";
/**
 * Klaviyo API
 * The Klaviyo REST API. Please visit https://developers.klaviyo.com for more details.
 *
 * Contact: developers@klaviyo.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ProfilePredictiveAnalyticsChannelAffinityPriorityCondition = void 0;
class ProfilePredictiveAnalyticsChannelAffinityPriorityCondition {
    static getAttributeTypeMap() {
        return ProfilePredictiveAnalyticsChannelAffinityPriorityCondition.attributeTypeMap;
    }
}
exports.ProfilePredictiveAnalyticsChannelAffinityPriorityCondition = ProfilePredictiveAnalyticsChannelAffinityPriorityCondition;
ProfilePredictiveAnalyticsChannelAffinityPriorityCondition.attributeTypeMap = [
    {
        "name": "type",
        "baseName": "type",
        "type": "ProfilePredictiveAnalyticsEnum"
    },
    {
        "name": "dimension",
        "baseName": "dimension",
        "type": "ProfilePredictiveAnalyticsChannelAffinityPriorityCondition.DimensionEnum"
    },
    {
        "name": "measurement",
        "baseName": "measurement",
        "type": "PriorityEnum"
    },
    {
        "name": "predictedChannel",
        "baseName": "predicted_channel",
        "type": "ProfilePredictiveAnalyticsChannelAffinityPriorityCondition.PredictedChannelEnum"
    },
    {
        "name": "filter",
        "baseName": "filter",
        "type": "ProfilePredictiveAnalyticsChannelAffinityPriorityFilter"
    }
];
(function (ProfilePredictiveAnalyticsChannelAffinityPriorityCondition) {
    let DimensionEnum;
    (function (DimensionEnum) {
        DimensionEnum[DimensionEnum["ChannelAffinity"] = 'channel_affinity'] = "ChannelAffinity";
    })(DimensionEnum = ProfilePredictiveAnalyticsChannelAffinityPriorityCondition.DimensionEnum || (ProfilePredictiveAnalyticsChannelAffinityPriorityCondition.DimensionEnum = {}));
    let PredictedChannelEnum;
    (function (PredictedChannelEnum) {
        PredictedChannelEnum[PredictedChannelEnum["Email"] = 'email'] = "Email";
        PredictedChannelEnum[PredictedChannelEnum["Push"] = 'push'] = "Push";
        PredictedChannelEnum[PredictedChannelEnum["Sms"] = 'sms'] = "Sms";
    })(PredictedChannelEnum = ProfilePredictiveAnalyticsChannelAffinityPriorityCondition.PredictedChannelEnum || (ProfilePredictiveAnalyticsChannelAffinityPriorityCondition.PredictedChannelEnum = {}));
})(ProfilePredictiveAnalyticsChannelAffinityPriorityCondition = exports.ProfilePredictiveAnalyticsChannelAffinityPriorityCondition || (exports.ProfilePredictiveAnalyticsChannelAffinityPriorityCondition = {}));
