// klaviyo-api/dist/model/profilePredictiveAnalyticsChannelAffinityRankFilter.js
"use strict";
/**
 * Klaviyo API
 * The Klaviyo REST API. Please visit https://developers.klaviyo.com for more details.
 *
 * Contact: developers@klaviyo.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ProfilePredictiveAnalyticsChannelAffinityRankFilter = void 0;
class ProfilePredictiveAnalyticsChannelAffinityRankFilter {
    static getAttributeTypeMap() {
        return ProfilePredictiveAnalyticsChannelAffinityRankFilter.attributeTypeMap;
    }
}
exports.ProfilePredictiveAnalyticsChannelAffinityRankFilter = ProfilePredictiveAnalyticsChannelAffinityRankFilter;
ProfilePredictiveAnalyticsChannelAffinityRankFilter.attributeTypeMap = [
    {
        "name": "type",
        "baseName": "type",
        "type": "StringEnum"
    },
    {
        "name": "operator",
        "baseName": "operator",
        "type": "ProfilePredictiveAnalyticsStringFilterOperator"
    },
    {
        "name": "value",
        "baseName": "value",
        "type": "ProfilePredictiveAnalyticsChannelAffinityRankFilter.ValueEnum"
    }
];
(function (ProfilePredictiveAnalyticsChannelAffinityRankFilter) {
    let ValueEnum;
    (function (ValueEnum) {
        ValueEnum[ValueEnum["High"] = 'high'] = "High";
        ValueEnum[ValueEnum["Low"] = 'low'] = "Low";
        ValueEnum[ValueEnum["Medium"] = 'medium'] = "Medium";
    })(ValueEnum = ProfilePredictiveAnalyticsChannelAffinityRankFilter.ValueEnum || (ProfilePredictiveAnalyticsChannelAffinityRankFilter.ValueEnum = {}));
})(ProfilePredictiveAnalyticsChannelAffinityRankFilter = exports.ProfilePredictiveAnalyticsChannelAffinityRankFilter || (exports.ProfilePredictiveAnalyticsChannelAffinityRankFilter = {}));
