// klaviyo-api/dist/model/pushTokenUnregisterQueryResourceObjectAttributes.js
"use strict";
/**
 * Klaviyo API
 * The Klaviyo REST API. Please visit https://developers.klaviyo.com for more details.
 *
 * Contact: developers@klaviyo.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PushTokenUnregisterQueryResourceObjectAttributes = void 0;
class PushTokenUnregisterQueryResourceObjectAttributes {
    static getAttributeTypeMap() {
        return PushTokenUnregisterQueryResourceObjectAttributes.attributeTypeMap;
    }
}
exports.PushTokenUnregisterQueryResourceObjectAttributes = PushTokenUnregisterQueryResourceObjectAttributes;
PushTokenUnregisterQueryResourceObjectAttributes.attributeTypeMap = [
    {
        "name": "token",
        "baseName": "token",
        "type": "string"
    },
    {
        "name": "platform",
        "baseName": "platform",
        "type": "PushTokenUnregisterQueryResourceObjectAttributes.PlatformEnum"
    },
    {
        "name": "vendor",
        "baseName": "vendor",
        "type": "PushTokenUnregisterQueryResourceObjectAttributes.VendorEnum"
    },
    {
        "name": "profile",
        "baseName": "profile",
        "type": "PushTokenUnregisterQueryResourceObjectAttributesProfile"
    }
];
(function (PushTokenUnregisterQueryResourceObjectAttributes) {
    let PlatformEnum;
    (function (PlatformEnum) {
        PlatformEnum[PlatformEnum["Android"] = 'android'] = "Android";
        PlatformEnum[PlatformEnum["Ios"] = 'ios'] = "Ios";
    })(PlatformEnum = PushTokenUnregisterQueryResourceObjectAttributes.PlatformEnum || (PushTokenUnregisterQueryResourceObjectAttributes.PlatformEnum = {}));
    let VendorEnum;
    (function (VendorEnum) {
        VendorEnum[VendorEnum["Apns"] = 'apns'] = "Apns";
        VendorEnum[VendorEnum["Fcm"] = 'fcm'] = "Fcm";
    })(VendorEnum = PushTokenUnregisterQueryResourceObjectAttributes.VendorEnum || (PushTokenUnregisterQueryResourceObjectAttributes.VendorEnum = {}));
})(PushTokenUnregisterQueryResourceObjectAttributes = exports.PushTokenUnregisterQueryResourceObjectAttributes || (exports.PushTokenUnregisterQueryResourceObjectAttributes = {}));
