// klaviyo-api/dist/model/reviewResponseDTO20240715ObjectResourceAttributes.js
"use strict";
/**
 * Klaviyo API
 * The Klaviyo REST API. Please visit https://developers.klaviyo.com for more details.
 *
 * Contact: developers@klaviyo.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ReviewResponseDTO20240715ObjectResourceAttributes = void 0;
class ReviewResponseDTO20240715ObjectResourceAttributes {
    static getAttributeTypeMap() {
        return ReviewResponseDTO20240715ObjectResourceAttributes.attributeTypeMap;
    }
}
exports.ReviewResponseDTO20240715ObjectResourceAttributes = ReviewResponseDTO20240715ObjectResourceAttributes;
ReviewResponseDTO20240715ObjectResourceAttributes.attributeTypeMap = [
    {
        "name": "email",
        "baseName": "email",
        "type": "string"
    },
    {
        "name": "status",
        "baseName": "status",
        "type": "ReviewResponseDTO20240715ObjectResourceAttributes.StatusEnum"
    },
    {
        "name": "verified",
        "baseName": "verified",
        "type": "boolean"
    },
    {
        "name": "reviewType",
        "baseName": "review_type",
        "type": "ReviewResponseDTO20240715ObjectResourceAttributes.ReviewTypeEnum"
    },
    {
        "name": "created",
        "baseName": "created",
        "type": "Date"
    },
    {
        "name": "updated",
        "baseName": "updated",
        "type": "Date"
    },
    {
        "name": "images",
        "baseName": "images",
        "type": "Array<string>"
    },
    {
        "name": "product",
        "baseName": "product",
        "type": "ReviewProductDTO"
    },
    {
        "name": "rating",
        "baseName": "rating",
        "type": "number"
    },
    {
        "name": "author",
        "baseName": "author",
        "type": "string"
    },
    {
        "name": "content",
        "baseName": "content",
        "type": "string"
    },
    {
        "name": "title",
        "baseName": "title",
        "type": "string"
    },
    {
        "name": "smartQuote",
        "baseName": "smart_quote",
        "type": "string"
    },
    {
        "name": "publicReply",
        "baseName": "public_reply",
        "type": "ReviewPublicReply"
    }
];
(function (ReviewResponseDTO20240715ObjectResourceAttributes) {
    let StatusEnum;
    (function (StatusEnum) {
        StatusEnum[StatusEnum["Featured"] = 'featured'] = "Featured";
        StatusEnum[StatusEnum["Pending"] = 'pending'] = "Pending";
        StatusEnum[StatusEnum["Published"] = 'published'] = "Published";
        StatusEnum[StatusEnum["Rejected"] = 'rejected'] = "Rejected";
        StatusEnum[StatusEnum["Unpublished"] = 'unpublished'] = "Unpublished";
    })(StatusEnum = ReviewResponseDTO20240715ObjectResourceAttributes.StatusEnum || (ReviewResponseDTO20240715ObjectResourceAttributes.StatusEnum = {}));
    let ReviewTypeEnum;
    (function (ReviewTypeEnum) {
        ReviewTypeEnum[ReviewTypeEnum["Question"] = 'question'] = "Question";
        ReviewTypeEnum[ReviewTypeEnum["Rating"] = 'rating'] = "Rating";
        ReviewTypeEnum[ReviewTypeEnum["Review"] = 'review'] = "Review";
    })(ReviewTypeEnum = ReviewResponseDTO20240715ObjectResourceAttributes.ReviewTypeEnum || (ReviewResponseDTO20240715ObjectResourceAttributes.ReviewTypeEnum = {}));
})(ReviewResponseDTO20240715ObjectResourceAttributes = exports.ReviewResponseDTO20240715ObjectResourceAttributes || (exports.ReviewResponseDTO20240715ObjectResourceAttributes = {}));
