// klaviyo-api/dist/model/sendWhatsAppActionData.js
"use strict";
/**
 * Klaviyo API
 * The Klaviyo REST API. Please visit https://developers.klaviyo.com for more details.
 *
 * Contact: developers@klaviyo.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SendWhatsAppActionData = void 0;
class SendWhatsAppActionData {
    constructor() {
        /**
        * Flow action status.
        */
        this['status'] = SendWhatsAppActionData.StatusEnum.Draft;
    }
    static getAttributeTypeMap() {
        return SendWhatsAppActionData.attributeTypeMap;
    }
}
exports.SendWhatsAppActionData = SendWhatsAppActionData;
SendWhatsAppActionData.attributeTypeMap = [
    {
        "name": "message",
        "baseName": "message",
        "type": "FlowWhatsApp"
    },
    {
        "name": "status",
        "baseName": "status",
        "type": "SendWhatsAppActionData.StatusEnum"
    }
];
(function (SendWhatsAppActionData) {
    let StatusEnum;
    (function (StatusEnum) {
        StatusEnum[StatusEnum["Draft"] = 'draft'] = "Draft";
        StatusEnum[StatusEnum["Live"] = 'live'] = "Live";
        StatusEnum[StatusEnum["Manual"] = 'manual'] = "Manual";
    })(StatusEnum = SendWhatsAppActionData.StatusEnum || (SendWhatsAppActionData.StatusEnum = {}));
})(SendWhatsAppActionData = exports.SendWhatsAppActionData || (exports.SendWhatsAppActionData = {}));
